driver: Add driver for Dediprog SPI-flash emulator

sjg20 opened the pull request:
The EM100-pro is a SPI-flash emulator which supports a variety of chips. It can be used to provide an image to a booting system.

Add a driver which allows images to be loaded ready for use. Provide a trace facility for use when the board does not boot.

This emulator is widely used on x86 machines since they boot using SPI flash. It is prohibitively expensive (in terms of time) to re-flash the SPI chip on test run, whereas the Dediprog emulator updates in about 5 seconds.

This has been tested on several x86 Chromebooks, including coral.

- [X] Documentation for the feature
- [ ] Tests for the feature 
- [X] The arguments and description in doc/configuration.rst have been updated
- [X] PR has been tested

sjg20 committed:
driver: Add driver for Dediprog SPI-flash emulator

The EM100-pro is a SPI-flash emulator which supports a variety of chips.
It can be used to provide an image to a booting system.

Add a driver which allows images to be loaded ready for use.

Signed-off-by: Simon Glass <sjg@chromium.org>

sjg20 commented:
This one needs variables, so I've removed the trace function in this version

obbardc commented on doc/configuration.rst:
```diff
@@ -1168,6 +1168,34 @@ NetworkDediprogFlasher
+overridden with a USB-connected emulator, which is much faster to load than
+reprogramming the DUT's actual SPI flash.
+
+So far only the Dediprog EM100-PRO is supported, so there is no 'model' property
```
Perhaps `model: Dediprog EM100-PRO` could be mandatory & checked in case of different models added in future?

sjg20 replied:
OK done

sjg20 commented:
I've added a test and dropped the 'model' property (since the driver only supports em100 and I'm not sure there is anything else it might need to support).
